crates/snapbox/src/assert.rs

```diff
@@ -201,13 +201,13 @@ impl Assert {
         // On `expected` being an error, make a best guess
         if let (Some(expected), format) = expected
             .as_ref()
-            .map(|d| (d.as_str(), d.format()))
-            .unwrap_or((Some(""), DataFormat::Text))
+            .map(|d| (d.render(), d.format()))
```

Comment:
This doesn't look correct.  I bet if you move the `if` removal first, this won't matter anymore